diff --git a/select.js b/select.js
index 148e8c50a86c11b1bf155a20190a4affb8f8dcbf..1f7515cae801baad0e8e1cb69696b2e7d63bb0c6 100644
--- a/select.js
+++ b/select.js
@@ -567,8 +567,8 @@ export class Select {
         this.checkboxes = [];
       }
       this.visibleOptions = this.checkboxes;
-
-      this.select.parentNode.parentNode.insertBefore(
+      
+      this.formGroup.insertBefore(
         this.selectMultiple,
         this.select.parentNode.nextSibling,
       );
