diff --git a/node_modules/@ecl/builder/bin/build.js b/node_modules/@ecl/builder/bin/build.js
index edd25b9..0c98c60 100755
--- a/node_modules/@ecl/builder/bin/build.js
+++ b/node_modules/@ecl/builder/bin/build.js
@@ -62,7 +62,7 @@ program
   .action(() => {
     const config = loadConfig(program.config);
     config.copy.forEach((conf) =>
-      copyFiles(conf.patterns || '**', conf.from, conf.to),
+      copyFiles(conf.from || "**", conf.to, conf.options),
     );
   });
 
diff --git a/node_modules/@ecl/builder/scripts/copy.js b/node_modules/@ecl/builder/scripts/copy.js
index 30019d2..30e38be 100644
--- a/node_modules/@ecl/builder/scripts/copy.js
+++ b/node_modules/@ecl/builder/scripts/copy.js
@@ -1,23 +1,32 @@
-const fs = require('fs');
-const path = require('path');
-const { promisify } = require('util');
-const { ncp } = require('ncp');
-const globby = require('globby');
+/**
+ * Copy files.
+ *
+ * @param {string} from - Path to a folder or file.
+ * @param {string} to - String to prefix, suffix or replace the current file name.
+ * @param {object} options - Object
+ *
+ * Example config object: {
+ *  copy: [
+ *   {
+ *     from: [path.resolve(nodeModules, "myfile.js")],
+ *     to: path.resolve(outputFolder, "js"),
+ *     options: { up: true },
+ *   },
+ * ],
+ */
  
-const copy = promisify(ncp);
 
-module.exports = (patterns, from, to) => {
-  const executor = async () => {
-    const paths = await globby(patterns, { nodir: true, cwd: from });
-
-    paths.map(async (file) => {
-      const input = path.resolve(from, file);
-      const dest = path.resolve(to, file);
+const fs = require("fs");
+const path = require("path");
+const copy = require("copyfiles");
  
-      fs.mkdirSync(path.dirname(dest), { recursive: true });
-      await copy(input, dest);
-    });
+module.exports = (from, to, options) => {
+  const executor = async () => {
+    from.push(to);
+    await copy(from, options, function (err) {
+      if (err) console.error(err);
+    })
   };
 
   executor();
-};
+}
