diff --git a/carousel.js b/carousel.js
index f37281da7b79e6e5f0b3622a6b4dae8802fc5d69..6b799b9effc841f584b95aafb7742a58a4b243ff 100644
--- a/carousel.js
+++ b/carousel.js
@@ -136,6 +136,8 @@ export class Carousel {
     this.direction = getComputedStyle(this.element).direction;
 
     this.slides = queryAll(this.slideClass, this.element);
+    this.slidesContainer.innerHTML = '';
+    this.slidesContainer.append(...this.slides);
     this.total = this.slides.length;
 
     // If only one slide, don't initialize carousel and hide controls
