diff --git a/mega-menu.js b/mega-menu.js
index 5d87032d4a96bb608cd008ad6b28a8613422d76c..98ae41567e2611f33e45e928a273129745196996 100644
--- a/mega-menu.js
+++ b/mega-menu.js
@@ -538,14 +538,14 @@ export class MegaMenu {
           this.element,
         );
         if (subItemExpanded) {
-          subItemExpanded.firstChild.classList.add(
+          queryOne(this.subLinkSelector, subItemExpanded).classList.add(
             'ecl-mega-menu__parent-link',
           );
         }
         const menuItem = this.openPanel.item;
         // Hide siblings
-        const siblings = menuItem.parentNode.childNodes;
-        siblings.forEach((sibling) => {
+        const siblings = menuItem.parentNode.children;
+        [...siblings].forEach((sibling) => {
           if (sibling !== menuItem) {
             sibling.style.display = 'none';
           }
@@ -555,7 +555,7 @@ export class MegaMenu {
       // Reset styles for the sublist and subitems
       subLists.forEach((list) => {
         list.classList.remove('ecl-mega-menu__sublist--scrollable');
-        list.childNodes.forEach((item) => {
+        [...list.children].forEach((item) => {
           item.style.display = '';
         });
       });
@@ -571,7 +571,7 @@ export class MegaMenu {
         this.element,
       );
       if (currentItem) {
-        currentItem.firstElementChild.classList.remove(
+        queryOne(this.subLinkSelector, currentItem).classList.remove(
           'ecl-mega-menu__parent-link',
         );
         currentItems.push(currentItem);
@@ -600,7 +600,7 @@ export class MegaMenu {
         this.element,
       );
       if (currentSubItem) {
-        currentSubItem.firstElementChild.classList.remove(
+        queryOne(this.subLinkSelector, currentSubItem).classList.remove(
           'ecl-mega-menu__parent-link',
         );
       }
@@ -743,7 +743,7 @@ export class MegaMenu {
                 .classList.contains('ecl-mega-menu__item--one-level-only')
             ) {
               if (items.length > 0) {
-                Array.from(items).forEach((item) => {
+                [...items].forEach((item) => {
                   itemsHeight += item.getBoundingClientRect().height;
                 });
               }
@@ -761,7 +761,7 @@ export class MegaMenu {
           );
           if (featuredPanelFirst) {
             // Get the elements inside the scrollable container and calculate their heights.
-            Array.from(featuredPanelFirst.firstElementChild.children).forEach(
+            [...featuredPanelFirst.firstElementChild.children].forEach(
               (child) => {
                 const elStyle = window.getComputedStyle(child);
 
@@ -790,7 +790,7 @@ export class MegaMenu {
           featuredPanel = queryOne('.ecl-mega-menu__featured', expanded);
           if (featuredPanel) {
             // Get the elements inside the scrollable container and calculate their heights.
-            Array.from(featuredPanel.firstElementChild.children).forEach(
+            [...featuredPanel.firstElementChild.children].forEach(
               (child) => {
                 const elStyle = window.getComputedStyle(child);
                 const marginHeight =
@@ -836,7 +836,7 @@ export class MegaMenu {
           mainPanel.style.height = `${height}px`;
           const seeAll = queryOne('.ecl-mega-menu__see-all', mainPanel);
           const firstOnly = mainPanel
-            .closest('li')
+            .closest('.ecl-mega-menu__item')
             .classList.contains('ecl-mega-menu__item--one-level-only');
           if (seeAll && firstOnly) {
             const remaining =
@@ -1329,9 +1329,9 @@ export class MegaMenu {
       const itemLink = queryOne(this.subLinkSelector, level2);
       itemLink.setAttribute('aria-expanded', 'false');
       itemLink.classList.remove('ecl-mega-menu__parent-link');
-      const siblings = level2.parentElement.childNodes;
+      const siblings = level2.parentElement.children;
       if (siblings) {
-        siblings.forEach((sibling) => {
+        [...siblings].forEach((sibling) => {
           sibling.style.display = '';
         });
       }
@@ -1515,17 +1515,17 @@ export class MegaMenu {
         });
 
         this.openPanel = { num: 2, item: menuItem };
-        siblings = menuItem.parentNode.childNodes;
+        siblings = menuItem.parentNode.children;
         if (this.isDesktop) {
           // Reset style for the siblings, in case they were hidden
-          siblings.forEach((sibling) => {
+          [...siblings].forEach((sibling) => {
             if (sibling !== menuItem) {
               sibling.style.display = '';
             }
           });
         } else {
           // Hide other items in the sublist
-          siblings.forEach((sibling) => {
+          [...siblings].forEach((sibling) => {
             if (sibling !== menuItem) {
               sibling.style.display = 'none';
             }
@@ -1577,7 +1577,7 @@ export class MegaMenu {
    */
   handleClickOnItem(e) {
     let isInTheContainer = false;
-    const menuItem = e.target.closest('li');
+    const menuItem = e.target.closest('.ecl-mega-menu__item');
 
     const container = queryOne(
       '.ecl-mega-menu__mega-container-scrollable',
