diff --git a/node_modules/@ecl/vanilla-component-content-block/content-block.js b/node_modules/@ecl/vanilla-component-content-block/content-block.js
index e231bd7..72de3e9 100644
--- a/node_modules/@ecl/vanilla-component-content-block/content-block.js
+++ b/node_modules/@ecl/vanilla-component-content-block/content-block.js
@@ -77,7 +77,7 @@ export class ContentBlock {
     }
 
     this.title = queryOne(this.titleSelector, this.element);
-    this.linkEl = this.title ? queryOne('a', this.title) : false;
+    this.linkEl = this.title.nodeName === 'A' ? this.title : queryOne('a', this.title);
     if (this.linkEl) {
       this.picture.style.cursor = 'pointer';
       const img = queryOne('img', this.picture);
