diff --git a/accordion.js b/accordion.js
index 08c5b160fb93ae39d28c7c450d9fe03cbdf55649..cd410378234668c4b90c4a4e905e1eab9d23e749 100644
--- a/accordion.js
+++ b/accordion.js
@@ -138,14 +138,21 @@ export class Accordion {
     if (iconElement) {
       const useNode = queryOne('use', iconElement);
       if (useNode) {
-        const originalXlinkHref = useNode.getAttribute('xlink:href');
-        let newXlinkHref = '';
+        var originalXlinkHref = useNode.getAttribute('xlink:href');
+        if (!originalXlinkHref) {
+          originalXlinkHref = useNode.getAttribute('href');
+        }
+        var newXlinkHref = '';
         if (isExpanded) {
           newXlinkHref = originalXlinkHref.replace('minus', 'plus');
         } else {
           newXlinkHref = originalXlinkHref.replace('plus', 'minus');
         }
-        useNode.setAttribute('xlink:href', newXlinkHref);
+        if (useNode.getAttribute('xlink:href')) {
+          useNode.setAttribute('xlink:href', newXlinkHref);
+        } else {
+          useNode.setAttribute('href', newXlinkHref);
+        }
       }
     }
 
