diff --git a/select.js b/select.js
index 5f3f041b4aafd937a42d34015fe923bcf6a8f8fb..f07d29eb5883bca578c341f163856f111e1a5106 100644
--- a/select.js
+++ b/select.js
@@ -1,5 +1,5 @@
 import { queryOne } from '@ecl/dom-utils';
-import * as getSystem from '@ecl/builder/utils/getSystem';
+import getSystem from '@ecl/builder/utils/getSystem';
 import EventManager from '@ecl/event-manager';
 
 const system = getSystem();
@@ -591,7 +591,7 @@ export class Select {
       }
       this.visibleOptions = this.checkboxes;
 
-      this.select.parentNode.parentNode.insertBefore(
+      this.formGroup.insertBefore(
         this.selectMultiple,
         this.select.parentNode.nextSibling,
       );
