diff --git a/select.js b/select.js
index af9c468227e205e3714541c6872671e1725bc9f0..5696a1ada73466cec4300b11b5330f18bb20d451 100644
--- a/select.js
+++ b/select.js
@@ -495,7 +495,7 @@ export class Select {
     }
     this.visibleOptions = this.checkboxes;
 
-    this.select.parentNode.parentNode.insertBefore(
+    this.select.closest('.ecl-form-group').insertBefore(
       this.selectMultiple,
       this.select.parentNode.nextSibling,
     );
