diff --git a/carousel.js b/carousel.js
index bb0daf7ebc5287514e1c929d6a10e9da714ddda0..7e496d339c32933ec2161f336cb1b34357bf4182 100644
--- a/carousel.js
+++ b/carousel.js
@@ -67,7 +67,7 @@ export class Carousel {
     this.slides = null;
     this.btnPlay = null;
     this.btnPause = null;
-    this.index = 1;
+    this.index = 0;
     this.total = 0;
     this.allowShift = true;
     this.activeNav = null;
@@ -84,8 +84,6 @@ export class Carousel {
     this.navigation = null;
     this.controls = null;
     this.direction = 'ltr';
-    this.cloneFirstSLide = null;
-    this.cloneLastSLide = null;
     this.executionCount = 0;
     this.maxExecutions = 5;
     this.slideWidth = 0;
@@ -128,6 +126,8 @@ export class Carousel {
     this.currentSlide = queryOne(this.currentSlideClass, this.element);
 
     this.slides = queryAll(this.slideClass, this.element);
+    this.slidesContainer.innerHTML = '';
+    this.slidesContainer.append(...this.slides);
     this.total = this.slides.length;
 
     // If only one slide, don't initialize carousel and hide controls
@@ -141,29 +141,10 @@ export class Carousel {
       return false;
     }
 
-    // Start initializing carousel
-    const firstSlide = this.slides[0];
-    const lastSlide = this.slides[this.slides.length - 1];
-
-    // Clone first and last slide
-    this.cloneFirstSlide = firstSlide.cloneNode(true);
-    this.cloneLastSlide = lastSlide.cloneNode(true);
-    this.slidesContainer.appendChild(this.cloneFirstSlide);
-    this.slidesContainer.insertBefore(this.cloneLastSlide, firstSlide);
-
-    // Initialize the js for the two cloned slides
-    const cloneFirstBanner = new ECL.Banner(
-      this.cloneFirstSlide.firstElementChild,
-    );
-    const cloneLastBanner = new ECL.Banner(
-      this.cloneLastSlide.firstElementChild,
-    );
-
-    cloneFirstBanner.init();
-    cloneLastBanner.init();
-
-    // Refresh the slides variable after adding new cloned slides
-    this.slides = queryAll(this.slideClass, this.element);
+    this.slides.forEach((slide) => {
+      const banner = new ECL.Banner(slide.firstElementChild.firstElementChild);
+      banner.init();
+    });
 
     // Initialze pagination and navigation
     this.handleResize();
@@ -216,10 +197,6 @@ export class Carousel {
    * Destroy component.
    */
   destroy() {
-    if (this.cloneFirstSLide && this.cloneLastSLide) {
-      this.cloneFirstSLide.remove();
-      this.cloneLastSLide.remove();
-    }
     if (this.btnPlay) {
       this.btnPlay.replaceWith(this.btnPlay.cloneNode(true));
     }
@@ -271,16 +248,13 @@ export class Carousel {
       this.executionCount = 0;
       return;
     }
-    const heightValues = this.slides.map((slide) => {
+     
+    const heightValues = this.slides.map(function (slide) {
       const banner = queryOne('.ecl-banner', slide);
-      const bannerInstance = ECL.components.get(banner);
-      const ratio = bannerInstance.defaultRatio();
-      bannerInstance.setHeight(ratio);
-      const padding =
-        parseInt(banner.style.getPropertyValue('--banner-footer-height'), 10) ||
-        0;
+      const padding = parseInt(banner.style.getPropertyValue('--banner-footer-height'), 10) || 0;
       const height = parseInt(banner.style.height, 10);
       const totalHeight = height + padding;
+    
       if (banner.style.height === 'auto') {
         // FRONT-4970 - Always handle banner heights, even when set to auto
         return banner.offsetHeight;
@@ -292,34 +266,26 @@ export class Carousel {
       return totalHeight;
     });
 
-    const elementHeights = heightValues.filter(
-      (height) => height !== undefined,
-    );
-
-    const tallestElementHeight = Math.max(...elementHeights);
+    const elementHeights = heightValues.filter(function (height) {
+      return height !== undefined;
+    });
 
-    if (
-      elementHeights.length === this.slides.length &&
-      tallestElementHeight > 1
-    ) {
+    const tallestElementHeight = Math.max.apply(Math, elementHeights);
+    
+    if (elementHeights.length === this.slides.length && tallestElementHeight === 0 || tallestElementHeight > 1) {
       clearInterval(this.intervalId);
 
       this.executionCount = 0;
-      this.slides.forEach((slide) => {
-        let bannerImage = null;
-        let bannerVideo = null;
+      this.slides.forEach(function (slide) {
         const banner = queryOne('.ecl-banner', slide);
+        let bannerImage = null;
+        let bannerVideo = null;
         if (banner) {
           bannerImage = queryOne('img', banner);
           bannerVideo = queryOne('video', banner);
-          const footerHeight =
-            parseInt(
-              banner.style.getPropertyValue('--banner-footer-height'),
-              10,
-            ) || 0;
+          const footerHeight = parseInt(banner.style.getPropertyValue('--banner-footer-height'), 10) || 0;
           const newHeight = tallestElementHeight - footerHeight;
-          banner.style.height = `${newHeight}px`;
-          banner.style.aspectRatio = 'auto';
+          banner.style.height = newHeight + "px";
         }
         if (bannerImage) {
           bannerImage.style.aspectRatio = 'auto';
@@ -434,9 +400,17 @@ export class Carousel {
   shiftSlide(dir, stopAutoPlay) {
     if (this.allowShift) {
       if (typeof dir === 'number') {
-        this.index = dir;
+        this.index = dir - 1;
       } else {
-        this.index = dir === 'next' ? this.index + 1 : this.index - 1;
+        if (dir === 'next') {
+          if (this.total > this.index + 1) {
+            this.index = this.index + 1;
+          } else {
+            this.index = 0;
+          }
+        } else {
+          this.index - 1;
+        }
       }
       this.moveSlides(true);
     }
@@ -474,11 +448,8 @@ export class Carousel {
       }
     }
     // Update index
-    if (this.index === 0) {
-      this.index = this.total;
-    }
-    if (this.index === this.total + 1) {
-      this.index = 1;
+    if (this.index === this.total) {
+      this.index = 0;
     }
 
     // Move slide without transition to ensure infinity loop
@@ -486,7 +457,7 @@ export class Carousel {
 
     // Update pagination
     if (this.currentSlide) {
-      this.currentSlide.textContent = this.index;
+      this.currentSlide.textContent = this.index + 1;
     }
 
     // Update slides
@@ -510,7 +481,7 @@ export class Carousel {
     // Update navigation
     if (this.navigationItems) {
       this.navigationItems.forEach((nav, index) => {
-        if (this.index === index + 1) {
+        if (this.index === index) {
           nav.setAttribute('aria-current', 'true');
           nav.removeAttribute('tabindex', -1);
         } else {
