diff --git a/select.js b/select.js
index 6f1de565807f1c91e5ac87117d8b4c60cc383ae8..37e8a9fc241d40bc7e15e1a846812ce0ce8ab8d1 100644
--- a/select.js
+++ b/select.js
@@ -1,5 +1,5 @@
 import { queryOne } from '@ecl/dom-utils';
-import * as getSystem from '@ecl/builder/utils/getSystem';
+import * as getSystem from '@ecl/builder/utils/getSystem';
 import EventManager from '@ecl/event-manager';
 
 const system = getSystem();
@@ -568,7 +568,7 @@ export class Select {
       }
       this.visibleOptions = this.checkboxes;
 
-      this.select.parentNode.parentNode.insertBefore(
+      this.formGroup.insertBefore(
         this.selectMultiple,
         this.select.parentNode.nextSibling,
       );
